app/src/Talk/TalkController.php

```diff
@@ -475,15 +480,21 @@ public function unlinkSpeaker($eventSlug, $talkSlug, $username)
 
         $unlinkSpeakerUri = $talkUri . "/speakers/" . $userId;
 
-        try {
-            $talkApi->unlinkVerifiedSpeakerFromTalk($unlinkSpeakerUri);
-        } catch (Exception $e) {
-            $this->application->flash('error', $e->getMessage());
+        $isAdmin = $event->getCanEdit();
+        if (!$isAdmin && $talk->getStartDateTime()->format('U') > time()) {
+            try {
+                $talkApi->unlinkVerifiedSpeakerFromTalk($unlinkSpeakerUri);
+            } catch (Exception $e) {
+                $this->application->flash('error', $e->getMessage());
+                $this->application->redirect($url);
+            }
+
+            $this->application->flash('message', 'Speaker has been removed from this talk.');
+            $this->application->redirect($url);
+        } else {
```

Comment:
Somehow I actually do :smile:

Personally I feel it'd be "cleaner" to have the "intended" flow of the function in the first indentation-level. For this function that would mean actually unlinking the user. So when the user is not able to unlink that's not the intended flow and would therefore happen within an ```if```-block. But that's my personal thing and I'm absolutely happy with the functionality being available at all ;)